### PR TITLE
Minor fixes & doc

### PR DESCRIPTION
**Pull request by fulmicoton**
- Bugfix in the metastore: an index deletion should result in a contorl plane restarting depending on the nature of the error.
- Added mssing doc in the metastore.

### CODE REVIEW

File: quickwit/quickwit-control-plane/src/control_plane.rs [hunk 1]
```diff
@@ -253,8 +253,8 @@ impl Handler<DeleteIndexRequest> for ControlPlane {
     ) -> Result<Self::Reply, ActorExitStatus> {
         let index_uid: IndexUid = request.index_uid.clone().into();
 
-        if let Err(error) = self.metastore.delete_index(request).await {
-            return Ok(Err(ControlPlaneError::from(error)));
+        if let Err(metastore_error) = self.metastore.delete_index(request).await {
+            return convert_metastore_error(metastore_error);
```
Comment (fulmicoton) [inline quickwit/quickwit-control-plane/src/control_plane.rs:257]: that's a bugfix I believe. Without this, if we encounterred an error from the metastore, depending on the error, we could observe a divergence between the control plane model  and the metastore.

File: quickwit/quickwit-metastore/src/metastore/postgresql_metastore.rs [hunk 2]
```diff
@@ -1022,11 +1022,11 @@ impl MetastoreService for PostgresqlMetastore {
         &mut self,
         request: IndexMetadataRequest,
     ) -> MetastoreResult<IndexMetadataResponse> {
-        let response = if let Some(index_id) = &request.index_id {
-            index_opt(&self.connection_pool, index_id).await?
-        } else if let Some(index_uid) = &request.index_uid {
+        let response = if let Some(index_uid) = &request.index_uid {
```
Comment (fulmicoton) [inline quickwit/quickwit-metastore/src/metastore/postgresql_metastore.rs:1025]: I just inverted uid and id.

File: quickwit/quickwit-control-plane/src/control_plane.rs (hunk 1 above)
Comment (fmassot) [reply quickwit/quickwit-control-plane/src/control_plane.rs:257]: We should add a test on this I think. Can you open an issue and assigned it to me?

**Review by fmassot — APPROVED** [review]

File: quickwit/quickwit-control-plane/src/control_plane.rs (hunk 1 above)
Comment (fulmicoton) [reply quickwit/quickwit-control-plane/src/control_plane.rs:257]: created #4185

**Pull request merged**